### PR TITLE
Mastermind rework

### PR DESCRIPTION
**Pull request by luximus-hunter**
- also reworks datamanager

### CODE REVIEW

File: Assets/Scripts/Minigames/Mastermind/Mastermind.cs [hunk 1]
```diff
@@ -1,88 +1,334 @@
...
 
+        void Win()
+        {
+            HiddenCodeRow.SetActive(false);
+        }
+
+        void Lose()
+        {
+            HiddenCodeRow.SetActive(false);
+        }
```
Comment (MischaLuijken) [inline Assets/Scripts/Minigames/Mastermind/Mastermind.cs]: word dit nog uitgebreid of is dit duplicate code?

File: Assets/Scripts/Navigation/NavigationController.cs [hunk 2]
```diff
@@ -31,8 +31,8 @@ public class Step
         private void Start()
         {
             UpdateGuide();
-            mapManager.SwitchMap(DataManager.Instance.CurrentMap);
-            meshRend = Steps[DataManager.Instance.CurrentStep].line.GetComponent<MeshRenderer>();
+            mapManager.SwitchMap(DataManager.CurrentMap);
```
Comment (MischaLuijken) [inline Assets/Scripts/Navigation/NavigationController.cs:34]: mapManager naming convention?

**Review by MischaLuijken — CHANGES_REQUESTED** [review]
Naming conventions and comments on code

File: Assets/Scripts/Minigames/Mastermind/Mastermind.cs (hunk 1 above)
Comment (luximus-hunter) [reply Assets/Scripts/Minigames/Mastermind/Mastermind.cs]: Methods have been made unique

File: Assets/Scripts/Navigation/NavigationController.cs (hunk 2 above)
Comment (luximus-hunter) [reply Assets/Scripts/Navigation/NavigationController.cs:34]: not in the scope of this pr. navigation will be reworked on a later date

**Review by MischaLuijken — APPROVED** [review]
[https://youtu.be/dQw4w9WgXcQ](url)

**Pull request merged**